src/libLookup/Lookup.h

```diff
@@ -169,6 +169,7 @@ class Lookup : public Executable, public Broadcastable {
   // TODO: move the Get and ProcessSet functions to Synchronizer
   bool GetSeedPeersFromLookup();
   bool GetDSInfoFromSeedNodes();
+  bool FetchDSInfoLoop();
```

Comment:
This function is better to be located inside Synchronizer.h

Reply:
Synchronizer does not have access to mediator

Reply:
You can pass the mediator in as a variable.